helpers.py

```diff
@@ -0,0 +1,162 @@
+                per_session_trials_dict= json.load(per_session_trials)
+                per_session_trials_data = pd.Series(per_session_trials_dict) # output is a series where each element is a dictionary of values for a single trial.
+        except:
+            print('json files are empty for: ', session_settings_paths[session_idx] )
+        # extract 
+        # from session settings: temporalfreq, autorewd; 
+        # From trials data: trialoutcome, reactiontimes, TF, stimT, Stim1TF (baseline freq), Stim2TF (change freq).
+
+        # definition of columns for the data frame and their content in ditionary of lists with keys as column names:
+        mouse_id = session_settings_paths[session_idx].name.split('_')[1]
+        
+        session_date = session_settings_paths[session_idx].name.split('_')[2]
+        session_time = session_settings_paths[session_idx].name.split('_')[3]
+        time_stamp = session_date+session_time
+        baseline_temporal_freq = session_settings.temporalfreq
+        auto_rewd = session_settings.autorewd
+
+        for trial in np.arange(len(per_session_trials_data)):
+            # trial = 0 #comment out after loop structuring
+            all_mouse_session_dict['mouse_id'].append(mouse_id)
+            all_mouse_session_dict['time_stamp'].append(time_stamp)
+            trial_outcome = per_session_trials_data[trial]['trialoutcome']
+            all_mouse_session_dict['session_trial_outcome'].append(trial_outcome) # a single string value of FA, Hit, Ref, Miss, Gray, abort
+            if trial_outcome == 'Hit': trial_outcome = 'RT'
+            all_mouse_session_dict['reaction_time'].append(per_session_trials_data[trial]['reactiontimes'][trial_outcome]) # an individual float value from a dictionary of reaction times for each one of the keys for the possible trial outcomes detailed above. 
+            temporal_freq_vector = np.array(per_session_trials_data[trial]['TF']) # a list (turned into an array) of values for the temporal frequency from before grating started till the trial ended. 
+            temporal_freq_filter = temporal_freq_vector > 0
+            temporal_freq_vector = temporal_freq_vector[temporal_freq_filter]
+            temporal_freq_vector = temporal_freq_vector[::3]
+            all_mouse_session_dict['session_TF_vectors'].append(temporal_freq_vector)
+            all_mouse_session_dict['baseline_TF'].append(per_session_trials_data[trial]['Stim1TF']) # the value of TF around which the baseline frequency revolves. 
+            all_mouse_session_dict['change_size_TF'].append(per_session_trials_data[trial]['Stim2TF']) # the value of the change TF (change size out of 4,2,1.5,1.35,1.25)
+            all_mouse_session_dict['change_time'].append(per_session_trials_data[trial]['stimT'])
+            all_mouse_session_dict['auto_reward'].append(auto_rewd)
+    if return_dict_or_df != 'dict':
```

Comment:
In general I think this is sensible way of organising the data. I might have a bit of a preference for some compartmentalisation as mentioned before, e.g. a set of dictionaries where the main key is the subject, within that are dictionarys where the keys are sessions, sessions and within that are the trials database. Then for plotting you can generate dataframes on the fly based on the subjects to incude. 

This feels a little more flexible and avoids duplication in dataframe columns, personally I think it is always better to request the data subsets that you want explicitly that pull everything and exclude the things you don't want. However, this is edging towards the realm of personal preference and if this approach feels comfortable for you I would continue with it as there are no tangible issues with it. The only benefit in an alternative approach may be much further down the line when you have many subjects and sessions. There may be some performance issues of generating a large database including every subject /session, and then having to exclude large parts of it (say, if you wanted to plot the first 10 trials of the first 2 sessions of a group of 10 mice, if you might every need to do something like that :'D)